tools/prepare-source.sh

```diff
@@ -1,25 +1,32 @@
 #!/bin/sh
-set -eux
+set -eu
 
-GO_VERSION=1.22.0
+GO_MAJOR_VER=1.22
+GO_VERSION=1.22.12
 OAPI_VERSION=2.4.1
-
-GO_BINARY=$(go env GOPATH)/bin/go$GO_VERSION
 TOOLS_PATH="$(realpath "$(dirname "$0")/bin")"
 
-# Install Go SDK
-go install golang.org/dl/go$GO_VERSION@latest
-$GO_BINARY download
+# Check latest Go version for the minor we're using
+LATEST=$(curl -s https://endoflife.date/api/go/"${GO_MAJOR_VER}".json  | jq -r .latest)
+if test "$LATEST" != "go$GO_VERSION"; then
```

Comment:
This doesn't work, I see the WARNING even tho we are on latest, I believe the issue is:
```suggestion
if test "$LATEST" != "$GO_VERSION"; then
```

Reply:
Right, rebased.